ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/flat/FlatDbStrategy.java

```diff
@@ -117,6 +118,65 @@ public Optional<Bytes> getCode(
     }
   }
 
+  /*
+   * Puts the account data for the given account hash, using the world state root hash supplier and node loader.
+   */
+  public void putFlatAccount(
+      final SegmentedKeyValueStorageTransaction transaction,
+      final Hash accountHash,
+      final Bytes accountValue) {
+    transaction.put(ACCOUNT_INFO_STATE, accountHash.toArrayUnsafe(), accountValue.toArrayUnsafe());
+  }
+
+  public void removeFlatAccount(
+      final SegmentedKeyValueStorageTransaction transaction, final Hash accountHash) {
+    transaction.remove(ACCOUNT_INFO_STATE, accountHash.toArrayUnsafe());
+  }
+
+  /*
+   * Puts the storage value for the given account hash and storage slot key, using the world state root hash supplier, storage root supplier, and node loader.
+   */
+  public void putFlatAccountStorageValueByStorageSlotHash(
+      final SegmentedKeyValueStorageTransaction transaction,
+      final Hash accountHash,
+      final Hash slotHash,
+      final Bytes storage) {
+    transaction.put(
+        ACCOUNT_STORAGE_STORAGE,
+        Bytes.concatenate(accountHash, slotHash).toArrayUnsafe(),
+        storage.toArrayUnsafe());
+  }
+
+  /*
+   * Removes the storage value for the given account hash and storage slot key, using the world state root hash supplier, storage root supplier, and node loader.
+   */
+  public void removeFlatAccountStorageValueByStorageSlotHash(
+      final SegmentedKeyValueStorageTransaction transaction,
+      final Hash accountHash,
+      final Hash slotHash) {
+    transaction.remove(
+        ACCOUNT_STORAGE_STORAGE, Bytes.concatenate(accountHash, slotHash).toArrayUnsafe());
```

Comment:
why is it called ACCOUNT_STORAGE_STORAGE ? there's no ACCOUNT_STORAGE, right?

Reply:
yeah, not my favorite name, but it is what the segment enum calls it.  It is the underlying storage for account storage, but makes for a weird name.  I am all for renaming the flat segments to ACCOUNT_INFO_FLAT, ACCOUNT_STORAGE_FLAT, and CODE_STORAGE_FLAT or something else that is more meaningful.  I'd rather do that in another PR though since it would touch a lot of places